### ISSUE TITLE
Search Engines sometimes fail the tests on greatfirewallofchina

### ISSUE THREAD

**Issue by cqx931** [dhowe/ChinaEye#50]
It is not consistent that yahoo/bing are allowed/blocked... 
Currently if the search keyword matches the triggers, the page won't be checked and the menu displays the server info as OK. 
However, this would be very strange if the search engine url fails the server tests, which means that the server result displayed is 'ok' if the keywords is blocked, and 'fail' if the keyword is not blocked.

One way to avoid this is to always check server for the webpage.

**Comment by dhowe** [comment]
are any of the search engines actually blocked in china (according to GFW) ?

or are you saying that the results (from GFW) are inconsistent?

**Comment by cqx931** [comment]
1. There are some differences between the result I got from GFW and my personal experience/ my friend's live testing.
According to the testing of a friend in China, Google search is blocked, while Bing and yahoo.com/yahoo.tw are ok.
However when I test them on GFW, google has never been blocked so far, while bing and yahoo.tw sometimes is blocked.
Now when I check http://www.baidu.com on GFW, it fails the tests as well... which doesn't make sense to me at all.

2. GFW's own inconsistency
Bing used to be able to pass the server test, but from the result I got today and yesterday, it is not.
I do believe that the blocking situation changes in China, but the result from GFW doesn't seem to  change completely according to the situation.

3. Regional difference
According to the same friend, yahoo.hk is blocked while yahoo.tw is not.

My friend is currently in Yunnan, which is the last server area that is supported by GFW.

**Comment by cqx931** [comment]
Since there are cases like No.2 or No.3, I think it is better to not pass the server tests if a search keyword matches the triggers. So that ChinaEye can keep consistence in it's own result.

**Comment by dhowe** [comment]
you mean that if, the search is a trigger word, we dont check GFW, otherwise we do?

**Comment by cqx931** [comment]
> you mean that if, the search is a trigger word, we dont check GFW, otherwise we do?

This is our current approach
I am suggesting to always check GFW, even after a trigger word is detected

**Comment by cqx931** [comment]
@dhowe what's your opinion on this? Both in the current result from GFW or the actually situation, the search engine we covered are now always allowed. So it makes sense to always check both the search keyword and the search result.
And the blocking status should be set after checking keyword and server.

This will also related to https://github.com/dhowe/ChinaEye/issues/59.

**Comment by dhowe** [comment]
> And the blocking status should be set after checking keyword and server.

Makes sense... if either status is 'blocked' though, do we still need to check the other?

**Comment by cqx931** [comment]
It was ok not to check the other when we don't display them in the menu. But since now we are showing blocking status for both of them, it makes sense to check them both... or you mean it should only check the other if the menu is opened?

**Comment by dhowe** [comment]
Yes, but maybe it is more trouble than its worth

**Comment by cqx931** [comment]
yes... especially when we are getting the server info from the cache, it doesn't really save much effort to not check the blocking status of an searh engine.

**Comment by dhowe** [comment]
agreed
